lib/sqlalchemy/orm/strategies.py

```diff
@@ -235,6 +235,9 @@ def create_row_processor(
 class ExpressionColumnLoader(ColumnLoader):
     def __init__(self, parent, strategy_key):
         super(ExpressionColumnLoader, self).__init__(parent, strategy_key)
+        self.have_default_expression = any([
+            not c.expression.type._isnull for c in self.parent_property.columns
+        ])
```

Comment:
default expressions are tested only once at initialization, using a state have_default_expression to avoid checking everytime setup_query is called.